src/components/Settings/Settings.tsx

```diff
@@ -163,69 +109,11 @@ export class Settings extends Component<Props, State> {
         }
     };
 
-    toggleMenu = () => {
-        const { isOpen } = this.state;
-        this.setState({ isOpen: !isOpen }, () => {
-            if (isOpen) {
-                this.handleClose();
-            } else {
-                // handle drop down toggle click
-                document.addEventListener("click", this.handleExternalClick, true);
-                if (this.mainMenuRef) {
-                    (this.mainMenuRef.children[0] as HTMLElement).focus();
-                }
-            }
-        });
-    };
-
-    handleMenuIconKeyPressed = (e: KeyboardEvent) => {
-        if (e.key === "Enter" || e.key === " ") {
-            this.toggleMenu();
-        }
-    };
-
-    getResourceSettings = (type: ResourceTypes) => {
-        if (type === ResourceTypes.VIDEO) {
-            this.setState({
-                showCameraSettings: true,
-                showAudioSettings: false,
-                showScreenShareSettings: false
-            });
-        } else if (type === ResourceTypes.AUDIO) {
-            this.setState({
-                showCameraSettings: false,
-                showAudioSettings: true,
-                showScreenShareSettings: false
-            });
-        } else {
-            this.setState({
-                showCameraSettings: false,
-                showAudioSettings: false,
-                showScreenShareSettings: true
-            });
-        }
-    };
-
-    handleChooseDevice = (device: MediaDeviceInfo) => {
-        if (device.kind === "videoinput") {
-            this.setState({ selectedCamera: device }, () => {
-                this.saveSettings();
-            });
-        } else {
-            this.setState({ selectedAudio: device }, () => {
-                this.saveSettings();
-            });
-        }
-    };
-
     handleClose = () => {
         document.removeEventListener("click", this.handleExternalClick, true);
 
         this.setState({
-            isOpen: false,
-            showAudioSettings: false,
-            showCameraSettings: false,
-            showScreenShareSettings: false
+            showSettingsOf: false
```

Comment:
```suggestion
            this.setState((prevState) => {
              ...prevState,
              showSettingsOf: undefined
            });
```
to be on the safe side here, let's use the previous state and use a callback internal function to be excecuted.